src/Umbraco.Core/Exceptions/WontImplementException.cs

```diff
@@ -1,27 +1,52 @@
 ﻿using System;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.Exceptions
 {
     /// <summary>
     /// The exception that is thrown when a requested method or operation is not, and will not be, implemented.
     /// </summary>
-    /// <remarks>The <see cref="NotImplementedException"/> is to be used when some code is not implemented,
+    /// <remarks>
+    /// The <see cref="NotImplementedException" /> is to be used when some code is not implemented,
     /// but should eventually be implemented (i.e. work in progress) and is reported by tools such as ReSharper.
     /// This exception is to be used when some code is not implemented, and is not meant to be, for whatever
-    /// reason.</remarks>
+    /// reason.
+    /// </remarks>
+    /// <seealso cref="System.NotImplementedException" />
+    [Obsolete("If a method or operation is not, and will not be, implemented, it is invalid, so we should throw an InvalidOperationException instead.")]
```

Comment:
I've marked `WontImplementException` obsolete, as you should just throw `InvalidOperationException` instead.